jwst/ramp_fitting/tests/test_ramp_fit.py

```diff
@@ -403,6 +403,167 @@ def test_oneCR_10_groups_combination_noisy2ndSegment():
         #even with noiser second segment, final slope should be just the average since they have the same number of groups
     np.testing.assert_allclose(slopes.data[500, 500], avg_slope,rtol=1e-5)
 
+
+def test_twenty_groups_two_segments():
+    ''' Test to verify weighting of multiple segments in combination: 
+        a) gdq all 0 ; b) 1 CR (2 segments) c) 1 CR then SAT (2 segments)
+    '''
+    (ngroups, nints, nrows, ncols, deltatime) = (20, 1, 1, 3, 6.) 
+    model1, gdq, rnModel, pixdq, err, gain = setup_small_cube( ngroups, 
+        nints, nrows, ncols, deltatime)
+
+    # a) ramp having gdq all 0 
+    model1.data[0,:,0,0] = np.arange(ngroups)*10. + 30. 
+
+    # b) ramp having 1 CR at group 15; 2 segments
+    model1.data[0,:,0,1] = np.arange(ngroups)*10. + 50. 
+    gdq[0,15,0,1] = dqflags.group['JUMP_DET']
+    model1.data[0,15:,0,1] += 1000.
+
+    # c) ramp having 1 CR at group 2; SAT starting in group 15
+    model1.data[0,:,0,2] = np.arange(ngroups)*10. + 70. 
+    gdq[0,2,0,2] =  dqflags.group['JUMP_DET']
+    model1.data[0,2:,0,2] += 2000. 
+    gdq[0,15:,0,2] = dqflags.group['SATURATED']
+    model1.data[0,15:,0,2] = 25000. 
+
+    new_mod, int_model, opt_model, gls_opt_model = ramp_fit(model1, 1024*30000.,
+        True, rnModel, gain, 'OLS', 'optimal')
+
+    # Check some PRI & OPT output arrays
+    np.testing.assert_allclose( new_mod.data, 10./deltatime, rtol=1E-4 )
+
+    wh_data = opt_model.slope != 0. # only test existing segments
+    np.testing.assert_allclose(opt_model.slope[wh_data], 10./deltatime, rtol=1E-4)
+    np.testing.assert_allclose(opt_model.yint[0,0,0,:], model1.data[0,0,0,:], rtol=1E-5)
+    np.testing.assert_allclose(opt_model.pedestal[0,0,:], 
+        model1.data[0,0,0,:]-10., rtol=1E-5)
+
+
+def test_miri_all_sat():
+    ''' Test of all groups in all integrations being saturated; all output arrays 
+        (particularly variances) should be 0.
+    '''
+    (ngroups, nints, nrows, ncols, deltatime) = (3, 2, 2, 2, 6.) 
+    model1, gdq, rnModel, pixdq, err, gain = setup_small_cube(ngroups, 
+        nints, nrows, ncols, deltatime)
+
+    model1.groupdq[:, :, :, :] = dqflags.group['SATURATED']
+
+    new_mod, int_model, opt_model, gls_opt_model = ramp_fit(model1, 1024*30000., 
+        True, rnModel, gain, 'OLS', 'optimal')
+
+    # Check PRI output arrays
+    np.testing.assert_allclose( new_mod.data, 0.0, rtol=1E-5  )
```

Comment:
Absolute tolerance (atol) might be more appropriate than relative tolerance when the expected value is zero.